api/adapter/controller.go

```diff
@@ -152,7 +153,7 @@ func computeAdapterHash(data *AdapterInsertModel, verify bool) error {
 
 	out, err := json.Marshal(input)
 	if err != nil {
-		panic(err)
+		return err
```

Comment:
Minor. Rather than repeating the same error message everywhere where `computAdapterHash` is used, we could return it here.

```suggestion
		return fmt.Sprintf(failed to compute adapter hash: %w", err.Error())
```

And then just call as 
```go
err = computeAdapterHash(&payload, verify)
if err != nil {
    return c.Status(fiber.StatusInternalServerError).SendString(err.Error())
}
```